### PR TITLE
Use deep equality in calledWith() to allow object/array arguments to match

### LINKED ISSUE #37: Ways to make calledWith() use deep equality
Currently, tests fail when passing an object/array literal to `calledWith()`, since JavaScript compares object equality based on identity. That is, the following test will fail:

```CoffeeScript
spy = bond()
spy(foo: "bar")
expect spy.calledWith(foo: "bar")
```

The workaround I've had to use is via Underscore's `isEqual()`:

```CoffeeScript
spy = bond()
spy(foo: "bar")
expect _.isEqual(spy.calledArgs[0], foo: "bar")
```

This works, but then I don't get to use the more semantic `calledWith()`. 

Two ideas: 
- Include Underscore, and rewrite `calledWith()` as such:
```CoffeeScript
  spy.calledWith = (args...) ->
    return false if !spy.called
    lastArgs = spy.calledArgs[spy.called-1]
    _.isEqual(args, lastArgs) # And remove the arrayEqual() function lower down in bond.coffee
```
- Allow an equality-checking callback to be passed in to `calledWith()`, allowing the client to pass in `_.isEqual`, for instance.

Option 1 adds some heft to the library. But in a testing context, I'm less concerned. (Also, it'd allow you to use [`_.find`](http://underscorejs.org/#find), which would allow for a more elegant solution to #36.) Option 2 would force users to pass arguments into `calledWith` as an array, so that the second argument could always be the callback, which seems a bit clunky.

Thoughts?

### PR DESCRIPTION
**Pull request by jessepinho**
Closes #37.

I started down the road of copying Underscore's `isEqual` functionality into this library, but found that it was a bit more convoluted than expected. (For example, the recursive functions called by `_.isEqual` reference other Underscore functions.) I propose simply including Underscore in this project. You'll notice Underscore is not actually included in the built JS file; rather, Underscore is added to `package.json` and `require`ed in `bond.coffee`.

Thoughts?

### CODE REVIEW

**Comment by EndangeredMassa** [comment]
This general approach is fine with me. I don't think we really need to support globals anymore.

Can you remove the usage line from the README that says you can still use globals? `<script src="bond.js"> -> window.bond(...)`

**Comment by jessepinho** [comment]
This shouldn't affect any behavior related to globals, though, right...? (Unless you want me to also remove the `bond` variable from `window` on [this line](https://github.com/jessepinho/bond/blob/6baec451dcc6ed31a863d58f2552b96d86ccd27c/bond.coffee#L167)?)

**Comment by EndangeredMassa** [comment]
Line 1 won't work with globals `_ = require 'underscore'` because require doesn't exist.

**Comment by jessepinho** [comment]
@EndangeredMassa Ah, gotcha. Just removed that (as well as the `window.bond` line I referenced in my comment above) in 441cd38.

File: bond.coffee [hunk 1]
```diff
@@ -170,5 +164,4 @@ bond = (obj, property) ->
     'restore': restore
   }
 
-window.bond = bond if window?
 module.exports = bond if module?.exports
```
Comment (EndangeredMassa) [inline bond.coffee]: Can we change this to the following?

```coffeescript
module.exports = bond
```

**Comment by jessepinho** [comment]
Done.

**Comment by EndangeredMassa** [comment]
published in [2.0.0](https://github.com/EndangeredMassa/bond/releases/tag/v2.0.0)

**Pull request merged**